packages/library/src/__tests__/transaction-confirmation-test.ts

```diff
@@ -143,8 +144,7 @@ describe('waitForDurableNonceTransactionConfirmation', () => {
             transaction: transactionWithoutFeePayerSignature,
         });
         await expect(commitmentPromise).rejects.toThrow(
-            "Could not determine this transaction's signature. Make sure that the transaction " +
-                'has been signed by its fee payer.',
+            new SolanaError({ __code: SOLANA_ERROR__TRANSACTION_SIGNATURE_NOT_COMPUTABLE }),
```

Comment:
nit: could be nice to offer an alternative way of constructing these if no context is required.

```ts
new SolanaError(SOLANA_ERROR__TRANSACTION_SIGNATURE_NOT_COMPUTABLE)
```

Reply:
Interesting. The reason everything was balled up into a single object had to do with the way that TypeScript refines generics, but maybe if you don't have any context this would be enough. I can go back into the trenches and try something like this.

Reply:
HA.

```ts
constructor(
    ...[code, context]: SolanaErrorContext[TErrorCode] extends undefined
        ? [code: TErrorCode]
        : [code: TErrorCode, context: SolanaErrorContext[TErrorCode]]
)
```